dom.bs

```diff
@@ -1151,21 +1151,16 @@ and an <a>event listener</a> <var>listener</var>, run these steps:
 
 <ol>
  <li>
-  <p>If <var>eventTarget</var> is a {{ServiceWorkerGlobalScope}} object, its associated
-  <a for="ServiceWorkerGlobalScope">service worker</a>'s <a for="service worker">script
-  resource</a>'s <a for="script resource">has ever been evaluated flag</a> is set, and
-  <var>listener</var>'s <a for="event listener">type</a> matches the {{Event/type}} attribute value
-  of any of the <a>service worker events</a>, then <a>report a warning to the console</a> with a
-  description explaining that the <a>service worker event</a> has to be added before the script
-  evaluation completes, otherwise the user agent might not start the <a>service worker</a> for the
-  <a>event</a>. [[!SERVICE-WORKERS]]
-
-  <p class="note no-backref">To avoid unnecessary delays, Service Workers [[!SERVICE-WORKERS]]
-  permits skipping event dispatch when no <a>event listeners</a> for the <a>service worker event</a>
-  have been deterministically added in the <a>service worker</a>’s <a for="service worker">set of
-  event types to handle</a> during the very first script evaluation. The <a>event listener</a> in
-  this step will still be added, but the implementations have to inform developers about this
-  behavior.
+  <p>If <var>eventTarget</var> is a {{ServiceWorkerGlobalScope}} object, its
+  <a for="ServiceWorkerGlobalScope">service worker</a>'s
+  <a for="service worker">script resource</a>'s
+  <a for="script resource">has ever been evaluated flag</a> is set, and <var>listener</var>'s
+  <a for="event listener">type</a> matches the {{Event/type}} attribute value of any of the
+  <a>service worker events</a>, then <a>report a warning to the console</a> that this might not give
+  the expected results. [[!SERVICE-WORKERS]]
+
+  <p class="note no-backref">Service workers are only started for event listeners added during the
+  first script evaluation.
```

Comment:
We have an occasion that is not covered by this note. Service Workers changed the skipping behavior to optional: https://w3c.github.io/ServiceWorker/#should-skip-event-algorithm step 1 uses "may". So, the listeners added after the first script evaluation still can see the service worker starts. Should we say: "Service workers may only be started for event listeners added during the first script evaluation?"

Reply:
Why was that changed?

Reply:
I ended up removing the note as the situation is complex and explaining it in detail here feels rather distracting for these general purpose methods.

Reply:
I think [the change](https://github.com/w3c/ServiceWorker/commit/2f7119069da24c095f6a6a8c47cd07f89d6bf143) was made to match the implementation. @mattto, can you give us more context on why you used *may* in that change?

Reply:
Yes, I think it was mostly to match the implementations. It was the consensus at the F2Fs according to https://github.com/w3c/ServiceWorker/issues/1200#issuecomment-342655003 and https://github.com/w3c/ServiceWorker/issues/1200#issuecomment-432951697.

Reply:
Limiting it to `fetch` seems okay, but making it random effectively requires user agents to look at each other's code of sites take a dependency. That's not great.

Reply:
I've reopened the issue to track this.